src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -1256,35 +1256,38 @@ private void EmitResource(ExpressionEmitter emitter, ImmutableArray<ExtensionExp
                     }
                 }
 
-                // Emit the options property 
-                if (resource.RetryOn is not null || resource.WaitUntil is not null)
+                // Emit the options property if there are entries in the DecoratorConfig dictionary
+                if (resource.DecoratorConfig.Count > 0)
                 {
                     emitter.EmitObjectProperty("@options", () =>
                     {
-                        if (resource.RetryOn is not null)
+                        if (resource.DecoratorConfig.ContainsKey("RetryOn"))
                         {
                             emitter.EmitArrayProperty("retryOn", () =>
                             {
-                                foreach (var item in resource.RetryOn.Items)
+                                foreach (var item in resource.DecoratorConfig["RetryOn"].Items)
                                 {
                                     emitter.EmitExpression(item);
                                 }
                             });
                         }
```

Comment:
Is it possible to make this code fully agnostic of each decorator?

E.g. something like:
```csharp
foreach (var (name, items) in resource.DecoratorConfig)
{
    emitter.EmitProperty(name, items);
}
```